cpp/src/arrow/flight/sql/server.cc

```diff
@@ -757,6 +788,29 @@ std::shared_ptr<Schema> SqlSchema::GetSqlInfoSchema() {
                               false)});
 }
 
+std::shared_ptr<Schema> SqlSchema::GetTypeInfoSchema() {
+  return arrow::schema({
+      field("type_name", utf8(), false),
+      field("data_type", int32(), false),
+      field("column_size", int32()),
+      field("literal_prefix", utf8()),
+      field("literal_suffix", utf8()),
+      field("create_params", list(utf8()), false),
```

Comment:
Maybe this should be `field("create_params", list(field("item", utf8(), false)))`?